### PR TITLE
impr: use tsrest for quotes endpoints (@fehmer)

### PR DESCRIPTION
**Pull request by fehmer**

### CODE REVIEW

File: backend/scripts/openapi.ts [hunk 1]
```diff
@@ -86,12 +86,19 @@ export function getOpenApi(): OpenAPIObject {
...
+
         {
           name: "psas",
           description: "Public service announcements.",
           "x-displayName": "PSAs",
           "x-public": "yes",
         },
+        {
+          name: "quotes",
+          description: "Quote submissions for new quotes and quote ratings.",
```
Comment (Miodec) [inline backend/scripts/openapi.ts:98]: "Quote ratings and new quote submissions"

File: packages/shared-types/src/index.ts [hunk 2]
```diff
@@ -48,6 +48,7 @@ export type PremiumInfo = {
   expirationTimestamp: number;
 };
 
+// Record<Language, Record<QuoteIdString, Rating>>
```
Comment (Miodec) [inline packages/shared-types/src/index.ts:51]: remove comment?

File: backend/src/api/routes/quotes.ts [hunk 3]
```diff
@@ -19,164 +15,61 @@ const checkIfUserIsQuoteMod = checkUserPermissions(["quoteMod"], {
...
-);
-
-const withCustomMessages = joi.string().messages({
-  "string.pattern.base": "Invalid parameter format",
+const s = initServer();
+export default s.router(quotesContract, {
+  get: {
+    middleware: [RateLimit.newQuotesGet],
+    handler: async (r) => callController(QuoteController.getQuotes)(r),
+  },
```
Comment (Miodec) [inline backend/src/api/routes/quotes.ts:23]: Missing checkIfUserIsQuoteMod

**Review by Miodec — CHANGES_REQUESTED** [review]

**Comment by fehmer** [comment]
this is a comment

**Review by Miodec — CHANGES_REQUESTED** [review]
test

**Review by Miodec — DISMISSED** [review]

**Review by Miodec — APPROVED** [review]
Thanks

**Pull request merged**